src/test/blake2s_test.cpp

```diff
@@ -27,37 +27,60 @@ TEST(TestG, TestTrue)
     ZERO.allocate(pb, "zero");
     pb.val(ZERO) = FieldT::zero();
 
-    // 0110 1011  0000 1000 1110 0110  0100 0111
     libsnark::pb_variable_array<FieldT> a = from_bits(
-        {0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0,
-         1, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1},
+        {
+            0, 1, 1, 0, 1, 0, 1, 1, // 6B
+            0, 0, 0, 0, 1, 0, 0, 0, // 08
+            1, 1, 1, 0, 0, 1, 1, 0, // E6
+            0, 1, 0, 0, 0, 1, 1, 1  // 47
+        },
         ZERO);
-    // 0101 0001  0000 1110  0101 0010  0111 1111
+
     libsnark::pb_variable_array<FieldT> b = from_bits(
-        {0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0,
-         0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1},
+        {
+            0, 1, 0, 1, 0, 0, 0, 1, // 51
+            0, 0, 0, 0, 1, 1, 1, 0, // 0E
+            0, 1, 0, 1, 0, 0, 1, 0, // 52
+            0, 1, 1, 1, 1, 1, 1, 1  // 7F
+        },
         ZERO);
-    // 0110 1010  0000 1001  1110 0110  0110 0111
+
     libsnark::pb_variable_array<FieldT> c = from_bits(
-        {0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1,
-         1, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1},
+        {
+            0, 1, 1, 0, 1, 0, 1, 0, // 6A
+            0, 0, 0, 0, 1, 0, 0, 1, // 09
+            1, 1, 1, 0, 0, 1, 1, 0, // E6
+            0, 1, 1, 0, 0, 1, 1, 1  // 67
+        },
         ZERO);
-    // 0101 0001  0000 1110  0101 0010  0111 0100
+
     libsnark::pb_variable_array<FieldT> d = from_bits(
-        {0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0,
-         0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0},
+        {
+            0, 1, 0, 1, 0, 0, 0, 1, // 51
+            0, 0, 0, 0, 1, 1, 1, 0, // 0E
+            0, 1, 0, 1, 0, 0, 1, 0, // 52
+            0, 1, 1, 1, 0, 1, 0, 0  // 74
+        },
         ZERO);
 
-    // 0110 1100  0110 1100  0110 0101  0110 1000
+    // "lleh"
```

Comment:
Where does this come from @rrtoledo? (same q for "ow o" below)

Reply:
This represents the first call of G_primitive in blake2s compression function when hashing "hello world".
If you remember blake2s compression function will split the input block into 32bit ("hell", "o wo", "rld") words in little endian ("lleh", "ow o", "dlr") and then process then 2 at a time ("lleh" and "ow o"). 

I will add these details here as the mention of "lleh" and "ow o" alone is confusing.

Reply:
`// The test correponds to the first call of the g_primitive of blake2s(b"hello world")
// As blake2s first format the input blocks in 32bit words in little endian,
// the inputs of the firs G_Primitive are "lleh" and "ow o"
// ("hello world" -> "hell" "o wo" "rld" plus padding-> "lleh" "ow o" "dlr" plus padding)
TEST(TestG, TestTrue)`

Reply:
> I will add these details here as the mention of "lleh" and "ow o" alone is confusing.

yes, thanks. Perfect, the comment you dropped right after clarifies this. we should take the habit to add comments where necessary so that the code base never becomes a "mystery" (and this saves time for when we come back these files in the future)